packages/mui-material/src/ButtonGroup/ButtonGroup.js

```diff
@@ -121,7 +121,9 @@ const ButtonGroupRoot = styled('div', {
         }),
       ...(ownerState.variant === 'text' &&
         ownerState.color !== 'inherit' && {
-          borderColor: alpha(theme.palette[ownerState.color].main, 0.5),
+          borderColor: theme.vars
+            ? `rgba(${theme.vars.palette[ownerState.color].main} / 0.5)`
```

Comment:
```suggestion
            ? `rgba(${theme.vars.palette[ownerState.color].mainChannel} / 0.5)`
```

Reply:
In a case like this, we transform the `main` color into a channel and attach it to the theme as `mainChannel`.

```js
// input
main: '#000000'

// output
main: '#000000'
mainChannel: '0 0 0'
```